stored_requests/events/api/api.go

```diff
@@ -0,0 +1,70 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/prebid/prebid-server/stored_requests/events"
+)
+
+type eventsAPI struct {
+	invalidations chan []string
+	updates       chan map[string]json.RawMessage
+}
+
+// NewEventsAPI creates an EventProducer that generates cache events from HTTP requests.
+// The returned httprouter.Handle must be registered on both POST (update) and DELETE (invalidate)
+// methods and provided an `:id` param via the URL, e.g.:
+//
+// apiEvents, apiEventsHandler, err := NewEventsApi()
+// router.POST("/stored_requests/:id", apiEventsHandler)
+// router.DELETE("/stored_requests/:id", apiEventsHandler)
+// listener := events.Listen(cache, apiEvents)
+//
+// The returned HTTP endpoint should not be exposed on a public network without authentication
+// as it allows direct writing to the cache via Update.
+func NewEventsAPI() (events.EventProducer, httprouter.Handle) {
+	api := &eventsAPI{
+		invalidations: make(chan []string),
+		updates:       make(chan map[string]json.RawMessage),
+	}
+	return api, httprouter.Handle(api.HandleEvent)
+}
+
+func (api *eventsAPI) HandleEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	id := ps.ByName("id")
+
+	if r.Method == "POST" {
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Missing config data.\n"))
+			return
+		}
+
+		// check if valid JSON
```

Comment:
We could go beyond just validating that it is JSON and validate the config itself, though this is intended to be an internal/authenticated endpoint and all requests would therefore come from the host/maintainers of the server instance.

Reply:
Should be fine either way.

In all honesty, I really doubt that anyone would use this in a prod setting. As soon as you deploy more than one instance, this would require the caller to be able to locate and call this endpoint on all of the instances. That's not generally easy in a containerized/microservice world, where the goal is usually to send a request to _an_ instance, not _all_ instances.

I suspect that this would be more like a development tool, in the same way that the `file_fetcher` is. It's definitely a convenient way to fiddle with the cache on a local instance.